web-app/js/unifina/streamr-table/streamr-table.js

```diff
@@ -62,30 +62,43 @@ StreamrTable.prototype.setHeaders = function(headers) {
     }
 }
 
-StreamrTable.prototype.addRow = function (row, rowId, op) {
-	if (op != "append") { op = "prepend" }
-	var newRow = $('<tr/>', {
+StreamrTable.prototype.addRow = function(row, rowId, op) {
+    if (op != 'append') {
+        op = 'prepend'
+    }
+    let newRow = $('<tr/>', {
         id: rowId != null ? rowId : '',
-	});
-	for (var i = 0; i < row.length; i++) {
-		var content = row[i] == null ? "" :
-					  row[i] instanceof Object ? JSON.stringify(row[i]) : row[i];
-		newRow.append($('<td/>', {
-			text: content,
-		}));
-	}
-	this.tableBody[op](newRow);
+    })
+    for (var i = 0; i < row.length; i++) {
+        let content
+        if (row[i] == null) {
+            content = ''
+        } else {
+            if (row[i] instanceof Object) {
+                if (row[i].date && row[i].streamr_date) {
```

Comment:
Where is `date` and/or `streamr_date` written? How come there are no changes in `VariadicEventTable.java` - you're still sending server-formatted strings from there!

**Here's how to get this on track:**

- Change this line to `if (row[i].__streamr_date) {` (lower probability of name collision with user objects)
- In `VariadicEventTable` line 48 (and line 39 in `EventTable`) add to the row a `Map` containing one item with key `__streamr_date` and the timestamp in milliseconds as value for that key.
```
Map<String, Object> dateObject = new HashMap<>();
dateObject.put("__streamr_date", getGlobals().time.getTime());
nr.add(dateObject);
```
- Remove the parsing, `content = new Date(row[i].__streamr_date)` should be enough